### DISCUSSION TITLE
Error: undefined method `[]' for nil:NilClass

### DISCUSSION THREAD

**Discussion by clarityD** [Homebrew/discussions#4366] [Casks]
### Output of `brew config`

```shell
/Users/Bryan
Bryan on Zeus command 822 : brew config
HOMEBREW_VERSION: 4.0.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: cf6614b50eac1c9f4bf386dc6989f0d443148df8
Last commit: 3 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bc2abb6032bbf1f70587e66299de0e1eb4e766bb
Core tap last commit: 36 minutes ago
Core tap branch: master
Core tap JSON: 30 Mar 10:27 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.ecFzSm51Mf/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit skylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 12.6.4-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
/Users/Bryan
Bryan on Zeus command 823 : brew doctor
Your system is ready to brew.
```

### Description of issue

History:

I have been using brew since 2016 and have completely removed and reinstalled it (after it changed from root install to standard user install). However, after I reinstalled it a few months ago, I discovered the cask for audacity never got "uninstalled". I got this error:
```
Error: undefined method `[]' for nil:NilClass
Please report this issue:
  https://docs.brew.sh/Troubleshooting
```

after the reinstall and every time I run:

`Bryan on Zeus command 819 : brew upgrade`

 and I have attempted to find a similar problem online to resolve this problem. But I didn't find anything that worked. =I installed audacity directly from https://www.audacityteam.org/download/mac/ at some point removing that may "fix" the issue. However the version I have installed is 2.3.2 and it seems unlikely. I should really upgrade audacity, it has not been high on my priority list.

I tried to reset Homebrew's cache:

`Bryan on Zeus command 833 : brew cleanup --prune=all`

and then force uninstall the cask:

`Bryan on Zeus command 834 :brew  uninstall --force --zap audacity`

but I keep getting: 

```
Error: undefined method `[]' for nil:NilClass
Please report this issue:
  https://docs.brew.sh/Troubleshooting

```
which is the same error I get every time I run:

`Bryan on Zeus command 819 : brew upgrade`

Thanks in advance!

### Relevant casks

audacity (2.1.2)

**Comment by carlocab** [comment]
What's the output of `brew  uninstall --force --zap --verbose --debug audacity`?

**Reply by clarityD** [reply]
Bryan on Zeus command 838 : brew uninstall --force --zap --verbose --debug audacity
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading audacity
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromPathLoader): loading /usr/local/Caskroom/audacity/.metadata/2.1.2/20161126054002.982/Casks/audacity.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading cowsay from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading libtool from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading coreutils from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading gmp from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading lolcat from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading gdbm from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading mpdecimal from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading cscope from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading libyaml from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading readline from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading sqlite from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading xz from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading unixodbc from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading lua from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading macvim from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading openssl@3 from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading ca-certificates from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading gettext from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading autoconf from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading m4 from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading openssl@1.1 from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading python@3.10 from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading python@3.11 from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading ruby from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading automake from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading fortune from API
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading audacity
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading gimp
==> Zapping Cask audacity
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromPathLoader): loading /usr/local/Caskroom/audacity/.metadata/2.1.2/20161126054002.982/Casks/audacity.rb
==> Implied `brew uninstall --cask audacity`
==> Uninstalling artifacts
==> 2 artifacts defined
Error: undefined method `[]' for nil:NilClass
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/usr/local/Caskroom/audacity/.metadata/2.1.2/20161126054002.982/Casks/audacity.rb:13:in `block (3 levels) in load'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open-uri.rb:169:in `open_uri'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open-uri.rb:736:in `open'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open-uri.rb:35:in `open'
/usr/local/Caskroom/audacity/.metadata/2.1.2/20161126054002.982/Casks/audacity.rb:11:in `block (2 levels) in load'
/usr/local/Homebrew/Library/Homebrew/cask/url.rb:118:in `instance_exec'
/usr/local/Homebrew/Library/Homebrew/cask/url.rb:118:in `call'
/usr/local/Homebrew/Library/Homebrew/cask/url.rb:191:in `block in initialize'
/usr/local/Homebrew/Library/Homebrew/lazy_object.rb:16:in `__getobj__'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/delegate.rb:97:in `respond_to_missing?'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/delegate.rb:82:in `respond_to?'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/delegate.rb:82:in `method_missing'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:641:in `inspect'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:641:in `inspect'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:641:in `inspect'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:641:in `inspect'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:641:in `inspect'
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:45:in `puts'
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:45:in `puts'
/usr/local/Homebrew/Library/Homebrew/extend/kernel.rb:45:in `odebug'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:458:in `uninstall_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:480:in `zap'
/usr/local/Homebrew/Library/Homebrew/cmd/uninstall.rb:76:in `block in uninstall'
/usr/local/Homebrew/Library/Homebrew/cmd/uninstall.rb:71:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/uninstall.rb:71:in `uninstall'
/usr/local/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'

**Comment by clarityD** [comment]
Just an update, I uninstalled the Audacity version 2.3.2 that I installed directly from the audacity website but that had no effect on the brew output.

**Reply by clarityD** [reply]
I'm think perhaps removing:

/usr/local/Caskroom/audacity/.metadata/2.1.2/20161126054002.982/Casks/audacity.rb

might fix the issue.? I would like to get it to a state where I can reinstall audacity from the brew cask so it stay updated. I have a backup, I think I'll give that a shot and see what happens.

**Reply by clarityD** [reply]
Bingo! Thanks carlocab! The debug info helped.

**Reply by carlocab** [reply]
Great; glad that helped.
